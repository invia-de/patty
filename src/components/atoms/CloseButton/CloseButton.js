import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../utils/classnames';

import styles from './CloseButton.module.scss';

export default function CloseButton({ className, onClick }) {
  return (
    <button
      type="button"
      aria-label="Close"
      className={cx(styles.button, className)}
      onClick={onClick}
    >
      <svg
        version="1.1"
        id="CloseButton"
        xmlns="http://www.w3.org/2000/svg"
        x="2px"
        y="2px"
        width="10px"
        height="10px"
        viewBox="0 0 512 512"
      >
        <path
          d="M255.973,358.455L102.457,511.971L0,409.518l153.515-153.516L0,102.485L102.457,0.029l153.516,153.516L409.49,0.029
	l102.455,102.456L358.429,256.002L512,409.518L409.49,511.971L255.973,358.455z"
        />
      </svg>
    </button>
  );
}
CloseButton.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired
};
