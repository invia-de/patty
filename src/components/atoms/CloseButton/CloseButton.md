```jsx
<CloseButton onClick={() => alert('Button clicked!')} />
```
