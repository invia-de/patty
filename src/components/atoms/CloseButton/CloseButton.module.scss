@import 'theme.scss';

$closebutton-background-color: $blue-dim !default;
$closebutton-color: $white !default;

.button {
  width: 18px;
  height: 18px;
  background-color: $closebutton-background-color;
  user-select: none;
  border: 0;
  color: $closebutton-color;
  padding: 3px;

  svg {
    display: block;
  }

  &:focus,
  &:hover {
    outline: 0;
  }
}
