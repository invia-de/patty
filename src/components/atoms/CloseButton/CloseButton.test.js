import React from 'react';
import CloseButton from './CloseButton';
import { render } from 'react-testing-library';

test('CloseButton renders', () => {
  const { container } = render(<CloseButton />);
  expect(container.firstChild).toMatchSnapshot();
});
