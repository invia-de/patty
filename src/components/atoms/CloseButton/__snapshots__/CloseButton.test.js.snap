// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders 1`] = `
<button
  aria-label="Close"
  class="button"
  type="button"
>
  <svg
    aria-hidden="true"
    class="icon"
    viewBox="0 0 512 512"
  >
    <path
      d="M255.973,358.455L102.457,511.971L0,409.518l153.515-153.516L0,102.485L102.457,0.029l153.516,153.516L409.49,0.029 l102.455,102.456L358.429,256.002L512,409.518L409.49,511.971L255.973,358.455z"
    />
  </svg>
</button>
`;

exports[`CloseButton renders with additional className 1`] = `
<button
  aria-label="Close"
  class="button test"
  type="button"
>
  <svg
    aria-hidden="true"
    class="icon"
    viewBox="0 0 512 512"
  >
    <path
      d="M255.973,358.455L102.457,511.971L0,409.518l153.515-153.516L0,102.485L102.457,0.029l153.516,153.516L409.49,0.029 l102.455,102.456L358.429,256.002L512,409.518L409.49,511.971L255.973,358.455z"
    />
  </svg>
</button>
`;
